library/ajax/addlistitem.php

```diff
@@ -50,27 +51,20 @@
 // determine the sequential order of the new item,
 // it should be the maximum number for the specified list plus one
 $seq = 0;
-$row = sqlQuery("SELECT max(seq) as maxseq FROM list_options WHERE list_id = '" . $list_id . "' AND activity = 1");
+$row = sqlQuery("SELECT max(seq) as maxseq FROM list_options WHERE list_id = '" . $list_id . "' AND activity = 1", array($list_id, $option_id);
 $seq = $row['maxseq']+1;
 
 // add the new list item
 $rc = sqlInsert("INSERT INTO list_options ( " .
-                "list_id, option_id, title, seq, is_default, option_value " .
-                ") VALUES (" .
-                "'".$list_id."'".
-                ",'".trim($option_id)."'" .
-                ",'".trim($title). "'" .
-                ",'".$seq."'" .
-                ",'".$is_default."'" .
-                ",'".$option_value."'".
-                ")");
+                "list_id, option_id, title, seq, is_default, option_value 
+                ) VALUES ( ?, ?, ?, ?, ?, ?", array($list_id, $option_id, $title, $seq, $is_default, $option_value);
 
 // return JSON data of list items on success
 echo '{ "error":"", "options": [';
 // send the 'Unassigned' empty variable
 echo '{"id":"","title":"' . xl('Unassigned') . '"}';
 $comma = ",";
-$lres = sqlStatement("SELECT * FROM list_options WHERE list_id = '$list_id' AND activity = 1 ORDER BY seq");
+$lres = sqlStatement("SELECT * FROM list_options WHERE list_id = ? AND activity = 1 ORDER BY ?", array($list_id, $seq);
```

Comment:
note that in above query `seq` is not a variable (so need to put the `seq` back at the end of query)